ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/numberinput/v1/numberinput/clientlibs/site/js/numberinputwidget.js

```diff
@@ -0,0 +1,408 @@
+/*******************************************************************************
+ * Copyright 2022 Adobe
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+class NumericInputWidget {
+    #widget=null
+    #model=null // passed by reference
+    #options=null
+    #defaultOptions={
+        value : null,
+        curValue: null,
+        pos: 0,
+        lengthLimitVisible: true,
+        zero:"0",
+        decimal:".",
+        minus:"-"
+    }
+    //TODO: to support writing in different locales \d should be replaced by [0-9] for different locales
+    #matchArray ={
+        "integer":"^[+-]?{digits}*$",
+        "decimal":"^[+-]?{digits}{leading}({decimal}{digits}{fraction})?$",
+        "float":"^[+-]?{digits}*({decimal}{digits}*)?$"
+    }
+    #regex=null
+    #processValue=null
+    #engRegex=null
+    #writtenInLocale=false
+    #previousCompositionVal=""
+
+    #toLatinForm    (halfOrFullWidthStr) {
+        // refer http://www.fileformat.info/info/unicode/block/halfwidth_and_fullwidth_forms/utf8test.htm
+        return halfOrFullWidthStr.replace(
+            /[\uff00-\uffef]/g,
+            function(ch) { return String.fromCharCode(ch.charCodeAt(0) - 0xfee0); }
+        );
+    }
+
+    constructor(widget, model) {
+        // initialize the widget and model
+        this.#widget = widget;
+        this.#model = model;
+        // initialize options for backward compatibility
+        this.#options = Object.assign({}, this.#defaultOptions, this.#model._jsonModel);
+        let matchStr =  this.#matchArray[this.#options.dataType];
+        if(matchStr) {
+            let ld = this.#options.leadDigits,
```

Comment:
a check of `decimal` data type should also come before extracting lead and frac digits.

Reply:
This is an existing code, I have not changed any logic.